src/cluster_asm.c

```diff
@@ -2517,11 +2583,19 @@ void asmTrimInBackground(slotRangeArray *slots) {
     sdsfree(str);
 }
 
+/* Trim the slots. */
 void asmTrimSlots(slotRangeArray *slots) {
     if (asmManager->debug_trim_method == ASM_DEBUG_TRIM_NONE)
         return;
-    /* TODO: Insert logic to select trim method */
-    asmTrimInBackground(slots);
+
+    /* TODO: Is that two event enough or shall we check for zero subscribers? */
+    int activetrim = (asmManager->debug_trim_method == ASM_DEBUG_TRIM_ACTIVE) ||
+                     (asmManager->debug_trim_method == ASM_DEBUG_TRIM_DEFAULT &&
+                      moduleHasSubscribersForKeyspaceEvent(NOTIFY_GENERIC | NOTIFY_TRIMMED));
+    if (activetrim)
+        asmTriggerActiveTrim(slots);
```

Comment:
in `asmTrimJobProcessPending`, we may try to start all `pending_trim_jobs`, but only one is running actually, and we propagate `trimslots` of all `pending_trim_jobs`, right?

Reply:
yes, I feel like it should be okay. Do you see any problem around here?

Reply:
I agree, `trim <key`> would have been no brainer. Now, we have all that logic of blocking master client because of this decision. I'm just hoping in mid term, we could drop active trimming and delete all this code.

Reply:
>  we have all that logic of blocking master client because of this decision.

yes. and module commands may have different return between master and replicas when active trimming

Reply:
> yes. and module commands may have different return between master and replicas when active trimming
currently, this is how it is in our fork. We trigger trimming separately. I feel safer for this concern. Though, I don't like all other complications.

Reply:
yep, add a new command

Reply:
OK, ignore